Modern/ops/src/test/java/org/bonej/ops/BoxIntersectTest.java

```diff
@@ -0,0 +1,337 @@
+ */
+public class BoxIntersectTest {
+
+	private static final ImageJ IMAGE_J = new ImageJ();
+	private static BinaryFunctionOp<ValuePair<Tuple3d, Vector3d>, Interval, Optional<ValuePair<DoubleType, DoubleType>>> boxIntersect;
+	private static Img<BitType> interval;
+
+	@Test
+	public void testCalculate() {
+		final Point3d origin = new Point3d(2, 2, -1);
+		final Vector3d direction = new Vector3d(0, 0, 1);
+		final ValuePair<Tuple3d, Vector3d> line = new ValuePair<>(origin,
+			direction);
+
+		final Optional<ValuePair<DoubleType, DoubleType>> result = boxIntersect
+			.calculate(line);
+		assertTrue(result.isPresent());
+		final ValuePair<DoubleType, DoubleType> scalars = result.get();
+		assertEquals(1.0, scalars.getA().get(), 1e-11);
+		assertEquals(6.0, scalars.getB().get(), 1e-11);
+	}
+
+	@Test
+	public void testEmptyInterval() {
+		final Point3d origin = new Point3d(0, 0, -1);
+		final Vector3d direction = new Vector3d(0, 0, 1);
+		final ValuePair<Tuple3d, Vector3d> line = new ValuePair<>(origin,
+			direction);
+		final Img<BitType> emptyInterval = ArrayImgs.bits(0, 0, 0);
+
+		final Optional<ValuePair<DoubleType, DoubleType>> result = boxIntersect
+			.calculate(line, emptyInterval);
+
+		assertFalse(result.isPresent());
+	}
+
+	@Test
+	public void testLineParallelToStackEdge() {
+		final Point3d origin = new Point3d(0, 0, -1);
+		final Vector3d direction = new Vector3d(0, 0, 1);
+		final ValuePair<Tuple3d, Vector3d> line = new ValuePair<>(origin,
+			direction);
+
+		final Optional<ValuePair<DoubleType, DoubleType>> result = boxIntersect
+			.calculate(line);
+
+		assertTrue(result.isPresent());
+		final ValuePair<DoubleType, DoubleType> scalars = result.get();
+		assertEquals(1.0, scalars.a.get(), 1e-11);
+		assertEquals(6.0, scalars.b.get(), 1e-11);
+	}
+
+	@Test
+	public void testCornerToCornerLine() {
+		final Point3d origin = new Point3d(-1, -1, -1);
+		final Vector3d direction = new Vector3d(1, 1, 1);
+		final ValuePair<Tuple3d, Vector3d> line = new ValuePair<>(origin,
+			direction);
+
+		final Optional<ValuePair<DoubleType, DoubleType>> result = boxIntersect
+			.calculate(line);
+
+		assertTrue(result.isPresent());
+		final ValuePair<DoubleType, DoubleType> scalars = result.get();
+		assertEquals(Math.sqrt(3), scalars.a.get(), 1e-11);
+		assertEquals(Math.sqrt(6 * 6 * 3), scalars.b.get(), 1e-11);
+	}
+
+	// Tests a line that only enters the interval for a short while
+	@Test
+	public void testShortSegment() {
+		final Point3d origin = new Point3d(0.5, 0.6, -0.5);
+		final Vector3d direction = new Vector3d(0, -1, 1);
+		final ValuePair<Tuple3d, Vector3d> line = new ValuePair<>(origin,
+			direction);
+
+		final Optional<ValuePair<DoubleType, DoubleType>> result = boxIntersect
+			.calculate(line);
+
+		assertTrue(result.isPresent());
+		final ValuePair<DoubleType, DoubleType> scalars = result.get();
+		assertEquals(Math.sqrt(2) / 2.0, scalars.a.get(), 1e-11);
+		assertEquals(0.848528137423857, scalars.b.get(), 1e-11);
+	}
```

Comment:
exact value is Math.sqrt(2)*0.6

Reply:
Cheers, fixed in e4a2cef1ea31376035d905d6793845b64b8eab4c